[[ Bug 22702 ]] Perform ad-hoc codesign of macOS standalones

livecodepanos opened the pull request.

livecodepanos commented:
WIP until we find out why the `it` variable is not empty:

```
2020-05-01 14:06:20.085651+0300 codesign[63728:2578109] [logging-persist] cannot open file at line 42270 of [95fbac39ba]
2020-05-01 14:06:20.085869+0300 codesign[63728:2578109] [logging-persist] os_unix.c:42270: (0) open(/var/db/DetachedSignatures) - Undefined error: 0
2020-05-01 14:06:20.208152+0300 codesign[63728:2578109] [logging-persist] cannot open file at line 42270 of [95fbac39ba]
2020-05-01 14:06:20.208181+0300 codesign[63728:2578109] [logging-persist] os_unix.c:42270: (0) open(/var/db/DetachedSignatures) - Undefined error: 0
```

livecodepanos commented:
> WIP until we find out why the `it` variable is not empty:
> 
> ```
> 2020-05-01 14:06:20.085651+0300 codesign[63728:2578109] [logging-persist] cannot open file at line 42270 of [95fbac39ba]
> 2020-05-01 14:06:20.085869+0300 codesign[63728:2578109] [logging-persist] os_unix.c:42270: (0) open(/var/db/DetachedSignatures) - Undefined error: 0
> 2020-05-01 14:06:20.208152+0300 codesign[63728:2578109] [logging-persist] cannot open file at line 42270 of [95fbac39ba]
> 2020-05-01 14:06:20.208181+0300 codesign[63728:2578109] [logging-persist] os_unix.c:42270: (0) open(/var/db/DetachedSignatures) - Undefined error: 0
> ```

It turns out this is probably just noise, so we could just use `the result` to check if the shell command was successful

montegoulding commented on ide-support/revsaveasstandalone.livecodescript:
```diff
@@ -731,11 +731,45 @@ command revSaveAsMacStandalone pStack, pRestoreFileName, pFolder, @xStandalonePa
    return empty
 end revSaveAsMacStandalone
 
+private command performAdHocCodesign pAppBundle
```
@livecodepanos probably better for `pAppBundle` to be either unescaped or changed to something like `pEscapedAppBundlePath` or something so someone doesn't come along later and add quotes or something.

runrevmark commented on ide-support/revsaveasstandalone.livecodescript:
```diff
@@ -731,11 +731,39 @@ command revSaveAsMacStandalone pStack, pRestoreFileName, pFolder, @xStandalonePa
+      local tAppBundle
+      set the itemDel to slash
+      put item 1 to -4 of tStandalonepath into tAppBundle
+      
```
Make sure you set the itemDelimeter back to comma after it is used, otherwise it might cause hard to spot errors in the future if more sections are added after this one.

runrevmark commented on ide-support/revsaveasstandalone.livecodescript:
```diff
@@ -731,11 +731,39 @@ command revSaveAsMacStandalone pStack, pRestoreFileName, pFolder, @xStandalonePa
+      throw "Clearing extended attributes failed with error:" && it
+   end if
+end clearExtendedAttributes
+
```
Only need one newline here

runrevmark reviewed (changes requested):
I suggest the commit message should be:
```
[[ Bug 22702 ]] Perform ad-hoc codesign of macOS standalones

This patch adds an ad-hoc code signature to macOS standalones after they have been
built. This is necessary because the unbound standalone engine is now signed so that
the IDE can be notarised. As binding a standalone invalidates the signature (as the
content of the executable changes), a replacement valid signature must be in place to
prevent errors when trying to run the resulting app on Catalina.
```

livecodepanos committed:
[[ Bug 22702 ]] Perform ad-hoc codesign of macOS standalones

This patch adds an ad-hoc code signature to macOS standalones after they have been
built. This is necessary because the unbound standalone engine is now signed so that
the IDE can be notarised. As binding a standalone invalidates the signature (as the
content of the executable changes), a replacement valid signature must be in place to
prevent errors when trying to run the resulting app on Catalina.

runrevmark commented:
@livecode-vulcan review ok 6bdd962

runrevmark commented:
@livecode-vulcan review ok 6bdd962

livecode-vulcan commented:
💙 review by @runrevmark ok 6bdd962ce7815743e2f6886bf2f9b2ba73c552f9

livecode-vulcan commented:
😞 test failure 6bdd962ce7815743e2f6886bf2f9b2ba73c552f9

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/794)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/203)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/203)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/203)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/203)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2320)
* try-community-universal-ios-iphoneos13.4: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.4/builds/53)
* try-community-universal-ios-iphonesimulator13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.4/builds/54)
* try-community-universal-mac-macosx10.9: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2348)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2336)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2335)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2370)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2310)

livecodepanos commented:
@livecode-vulcan retry 6bdd962

livecode-vulcan commented:
😎 test success 6bdd962ce7815743e2f6886bf2f9b2ba73c552f9

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/795)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/204)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/204)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/204)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/204)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2321)
* try-community-universal-ios-iphoneos13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.4/builds/54)
* try-community-universal-ios-iphonesimulator13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.4/builds/56)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2349)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2337)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2336)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2371)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2311)
